from src import *
from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_FileDecWindow(object):
    def setupUi(self, FileDecWindow):
        FileDecWindow.setObjectName("FileDecWindow")
        FileDecWindow.resize(500, 300)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("program\src\icon\icon.png"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        FileDecWindow.setWindowIcon(icon)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(FileDecWindow.sizePolicy().hasHeightForWidth())
        FileDecWindow.setSizePolicy(sizePolicy)
        self.centralwidget = QtWidgets.QWidget(FileDecWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.pw_entry = QtWidgets.QLineEdit(self.centralwidget)
        self.pw_entry.setGeometry(QtCore.QRect(110, 160, 281, 20))
        self.pw_entry.setText("")
        self.pw_entry.setMaxLength(32767)
        self.pw_entry.setFrame(True)
        self.pw_entry.setEchoMode(QtWidgets.QLineEdit.Normal)
        self.pw_entry.setDragEnabled(True)
        self.pw_entry.setCursorMoveStyle(QtCore.Qt.LogicalMoveStyle)
        self.pw_entry.setClearButtonEnabled(True)
        self.pw_entry.setObjectName("pw_entry")
        self.title_label = QtWidgets.QLabel(self.centralwidget)
        self.title_label.setGeometry(QtCore.QRect(10, 20, 481, 51))
        font = QtGui.QFont()
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        font.setKerning(True)
        self.title_label.setFont(font)
        self.title_label.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.title_label.setMouseTracking(False)
        self.title_label.setTextFormat(QtCore.Qt.AutoText)
        self.title_label.setAlignment(QtCore.Qt.AlignCenter)
        self.title_label.setObjectName("title_label")
        self.file_path_label = QtWidgets.QLabel(self.centralwidget)
        self.file_path_label.setGeometry(QtCore.QRect(110, 90, 240, 50))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.file_path_label.sizePolicy().hasHeightForWidth())
        self.file_path_label.setSizePolicy(sizePolicy)
        self.file_path_label.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.file_path_label.setAcceptDrops(False)
        self.file_path_label.setAlignment(QtCore.Qt.AlignCenter)
        self.file_path_label.setObjectName("file_path_label")
        self.file_select_btn = QtWidgets.QToolButton(self.centralwidget)
        self.file_select_btn.setGeometry(QtCore.QRect(360, 90, 30, 50))
        self.file_select_btn.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.file_select_btn.setPopupMode(QtWidgets.QToolButton.DelayedPopup)
        self.file_select_btn.setObjectName("file_select_btn")
        self.enc_progress_bar = QtWidgets.QProgressBar(self.centralwidget)
        self.enc_progress_bar.setEnabled(True)
        self.enc_progress_bar.setGeometry(QtCore.QRect(110, 220, 280, 50))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.enc_progress_bar.sizePolicy().hasHeightForWidth())
        self.enc_progress_bar.setSizePolicy(sizePolicy)
        self.enc_progress_bar.setAutoFillBackground(False)
        self.enc_progress_bar.setProperty("value", 24)
        self.enc_progress_bar.setTextVisible(False)
        self.enc_progress_bar.setOrientation(QtCore.Qt.Horizontal)
        self.enc_progress_bar.setInvertedAppearance(False)
        self.enc_progress_bar.setTextDirection(QtWidgets.QProgressBar.TopToBottom)
        self.enc_progress_bar.setObjectName("enc_progress_bar")
        self.file_enc_btn = QtWidgets.QPushButton(self.centralwidget)
        self.file_enc_btn.setGeometry(QtCore.QRect(110, 220, 280, 50))
        self.file_enc_btn.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.file_enc_btn.setObjectName("file_enc_btn")
        FileDecWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(FileDecWindow)
        QtCore.QMetaObject.connectSlotsByName(FileDecWindow)

    def retranslateUi(self, FileDecWindow):
        _translate = QtCore.QCoreApplication.translate
        FileDecWindow.setWindowTitle(_translate("FileDecWindow", "program_title"))
        self.pw_entry.setPlaceholderText(_translate("FileDecWindow", "Enter file password"))
        self.title_label.setText(_translate("FileDecWindow", "program_title"))
        self.file_path_label.setText(_translate("FileDecWindow", "Drag and drop file here"))
        self.file_select_btn.setText(_translate("FileDecWindow", "..."))
        self.file_enc_btn.setText(_translate("FileDecWindow", "File Decryption"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    FileDecWindow = QtWidgets.QMainWindow()
    ui = Ui_FileDecWindow()
    ui.setupUi(FileDecWindow)
    FileDecWindow.show()
    sys.exit(app.exec_())
